### ISSUE TITLE
Implicit curve plots may better use matplotlib contour

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#20056]
I have seen that sympy implicit plots are giving plots with inconsistent line thickness.
Maybe it is a wrong direction to use complicated adaptive sampling schemes to plot curves in the plane, while it can simply use matplotlib's contour feature with manual specification of levels.

```
from sympy import *

x, y = symbols('x y')
plot_implicit(x**3 + x**2 - y**2, (x, -2, 2), (y, -2, 2))
```
![image](https://user-images.githubusercontent.com/34944973/92338084-d2ec4580-f0e8-11ea-9c12-787b5d65d6b0.png)
```
import numpy as np
import matplotlib.pyplot as plt

f = lambdify([x, y], x**3 + x**2 - y**2)
xx = np.linspace(-2, 2, 100)
yy = np.linspace(-2, 2, 100)

xxx, yyy = np.meshgrid(xx, yy)
f(xxx, yyy)
plt.contour(xxx, yyy, f(xxx, yyy), [0.])
```
![image](https://user-images.githubusercontent.com/34944973/92338100-f911e580-f0e8-11ea-929d-951f66bf0944.png)

**Comment by Davide-sd** [comment]
The default implicit algorithm is good for plotting boolean expressions:

```
from sympy import *
x, y = symbols('x y')
plot_implicit(Eq(y, sin(x)) & (y > 0), (x, -7, 7), (y, -2, 2))
```

![image](https://user-images.githubusercontent.com/9921777/123633179-2c109b80-d819-11eb-8ad6-d7221204004e.png)

I think the implicit algorithm should only use the adaptive scheme only when dealing with Boolean expressions, otherwise it should use the traditional mesh grid approach.

**Comment by asmeurer** [comment]
We definitely need to improve this, although I wonder if there are ways the algorithm can be improved so that it handles curves like this better.

**Comment by asmeurer** [comment]
Also note that plot_implicit already has a flag `adaptive=False` that disables the interval arithmetic algorithm and uses the basic meshgrid method.

**Comment by sylee957** [comment]
No, `adaptive=False` still uses the method to draw small rectangles for line plots. The point I have is why should we stick to the method to draw small rectangles to make it look like a line instead of drawing contour lines directly

**Comment by Davide-sd** [comment]
No, when you set `adaptive=False` its going to use contours. You can verify this with the following.

```
p = plot_implicit(x**3 + x**2 - y**2, (x, -2, 2), (y, -2, 2), adaptive=True)
p._backend.ax[0].collections
# []
```

Here there are no contour lines. There must be a fill object somewhere, though I don't know where to look at. Whereas:

```
from sympy import *
var("x, y")
p = plot_implicit(x**3 + x**2 - y**2, (x, -2, 2), (y, -2, 2), adaptive=False)
p._backend.ax[0].collections

# [<matplotlib.collections.LineCollection at 0x7fec24e58a60>,
#  <matplotlib.collections.LineCollection at 0x7fec24e58c10>,
#  <matplotlib.collections.LineCollection at 0x7fec24e58d90>,
#  <matplotlib.collections.LineCollection at 0x7fec24e58fa0>,
#  <matplotlib.collections.LineCollection at 0x7fec24e589d0>,
#  <matplotlib.collections.LineCollection at 0x7fec24e2cd90>,
#  <matplotlib.collections.LineCollection at 0x7fec24e4c6d0>,
#  <matplotlib.collections.LineCollection at 0x7fec24b23220>,
#  <matplotlib.collections.LineCollection at 0x7fec24b23400>]
```

those are contour lines. The fact that those contours seems like little squares (hence horrible) depends on how it was coded. ImplicitSeries does something like this:

```
import numpy as np
import matplotlib.pyplot as plt

f = lambdify([x, y], x**3 + x**2 - y**2)
xx = np.linspace(-2, 2, 100)
yy = np.linspace(-2, 2, 100)

xxx, yyy = np.meshgrid(xx, yy)
zzz = f(xxx, yyy)
zzz[zzz < 0] = -1
zzz[zzz > 0] = 1
plt.contour(xxx, yyy, zzz)
# plt.contour(xxx, yyy, f(xxx, yyy), [0.])
```
![image](https://user-images.githubusercontent.com/9921777/123764154-d09be780-d8c4-11eb-8ea1-2c4b02f181c8.png)

**Comment by defencedog** [comment]
Its still present after 2 years
![8f021e97286942469a71d6ac594fc44a](https://user-images.githubusercontent.com/831749/196299805-ed687416-2ed5-442c-a9e2-93f0f86726a0.png)

**Comment by Davide-sd** [comment]
@defencedog You can try with [this module](https://github.com/Davide-sd/sympy-plot-backends):

```python
f = y-3*x+5 
g = x**2+y**2-5
h = y-x**3+3*x**2-3*x+1

plot_implicit(f, g, h, (x, -5, 5), (y, -5, 5), aspect="equal", \
              annotations=[{'xy': (-2, 2), 'text': '$g$'}, {'xy': (2, 4), 'text': '$h$'}, {'xy': (3.5, 4), 'text': '$f$'}], rendering_kw={"color": "b"} )
```
![index](https://user-images.githubusercontent.com/9921777/197884627-3d0c151a-271f-48d8-a43f-a31653ebcb74.png)

**Comment by defencedog** [comment]
@Davide-sd Your suggestion is HEAVY for my Android
